app/com/arpnetworking/metrics/portal/reports/impl/chrome/DevToolsServiceWrapper.java

```diff
@@ -66,18 +69,26 @@ public Object evaluate(final String js) {
     }
 
     @Override
-    public void navigate(final String url) {
+    public CompletionStage<Void> navigate(final String url) {
+        final CompletableFuture<Void> result = new CompletableFuture<>();
+        _dts.getPage().enable();
+        _dts.getPage().onLoadEventFired(e -> result.complete(null));
         _dts.getPage().navigate(url);
+        return result;
     }
 
     @Override
-    public void onLoad(final Runnable callback) {
-        _dts.getPage().enable();
-        _dts.getPage().onLoadEventFired(e -> callback.run());
+    public CompletionStage<Void> nowOrOnEvent(final String eventName, final Supplier<Boolean> ready) {
+        final CompletableFuture<Void> result = new CompletableFuture<>();
+        waitForEvent(eventName).thenAccept(foo -> result.complete(null));
+        if (ready.get()) {
```

Comment:
I'm confused; wont this block?

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get--